[CI] PointInTimeRelocationIT testPointInTimeRelocationPitPositionsInBcc failing

**Build Scans:**
- [elasticsearch-intake #42981 / part2](https://gradle-enterprise.elastic.co/s/6pedlygfmqgcs)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.recovery.PointInTimeRelocationIT.testPointInTimeRelocationPitPositionsInBcc" -Dtests.seed=84EA9DF105BB427A -Dtests.locale=dyo-SN -Dtests.timezone=Australia/Eucla -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.recovery.PointInTimeRelocationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testPointInTimeRelocationPitPositionsInBcc'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Closing pit should free open contexts. expected:<1> but was:<0>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testPointInTimeRelocationPitPositionsInBcc
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

> this will likely be fixed once https://github.com/elastic/elasticsearch/pull/152754 merges

Not really, unfortunately.
The error I see in the logs in the scan above that I think is causing the test failure is:

```
 1> [2026-07-07T17:09:03,672][WARN ][o.e.x.s.r.TransportStatelessUnpromotableRelocationAction][node_t1][generic][T#1] Unexpected exception while fetching Open PIT context info for shard [ycziisjybb][0], context id [P9-FbOOWTzaD_V21Q-CG6A][1] searcherId [okbugixjgr3VEeT1o4vcUA==]
  1> java.io.UncheckedIOException: java.io.IOException: Random IOException
  1> 	at org.elasticsearch.xpack.stateless.commits.BatchedCompoundCommit$BCCStatelessCompoundCommitsIterator.next(BatchedCompoundCommit.java:198)
  1> 	at org.elasticsearch.xpack.stateless.commits.BatchedCompoundCommit$BCCStatelessCompoundCommitsIterator.next(BatchedCompoundCommit.java:159)
  1> 	at org.elasticsearch.xpack.stateless.recovery.TransportStatelessUnpromotableRelocationAction.lambda$fetchOpenPitContextInfo$13(TransportStatelessUnpromotableRelocationAction.java:434)
  1> 	at org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:101)
  1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1114)
  1> 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
  1> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
  1> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
  1> 	at java.base/java.lang.Thread.run(Thread.java:1516)
  1> Caused by: java.io.IOException: Random IOException
  1> 	at org.elasticsearch.xpack.stateless.AbstractStatelessPluginIntegTestCase$1.failIfNeeded(AbstractStatelessPluginIntegTestCase.java:579)
  1> 	at org.elasticsearch.xpack.stateless.AbstractStatelessPluginIntegTestCase$1.blobContainerReadBlob(AbstractStatelessPluginIntegTestCase.java:606)
  1> 	at org.elasticsearch.xpack.stateless.StatelessMockRepository$StatelessMockBlobStore$StatelessMockBlobContainer.readBlob(StatelessMockRepository.java:129)
  1> 	at org.elasticsearch.xpack.stateless.objectstore.ObjectStoreService.lambda$readBatchedCompoundCommitFromStoreIncrementally$15(ObjectStoreService.java:932)
  1> 	at org.elasticsearch.xpack.stateless.commits.BatchedCompoundCommit$BCCStatelessCompoundCommitsIterator.next(BatchedCompoundCommit.java:187)
  1> 	... 8 more
```

It appears the mocked blob store randomly throws IOExceptions to simulate faulty network etc... When we catch this we don't relocate a pit but rather resume the recovery without it. I think whats wrong here is the assumption that we can expect a closed context in this case, because we haven't been able to create it in the first place.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -694,9 +694,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:exponential_histogram.allAggsInlineGrouped}
   issue: https://github.com/elastic/elasticsearch/issues/153123
-- class: org.elasticsearch.xpack.stateless.recovery.PointInTimeRelocationIT
-  method: testPointInTimeRelocationPitPositionsInBcc
-  issue: https://github.com/elastic/elasticsearch/issues/153149
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   method: test {csv-spec:spatial_shapes.convertCartesianShapeFromStringParseError}
   issue: https://github.com/elastic/elasticsearch/issues/153154

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/recovery/PointInTimeRelocationIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/recovery/PointInTimeRelocationIT.java
@@ -241,11 +241,7 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         updateIndexSettings(Settings.builder().put("index.routing.allocation.exclude._name", searchNodeCurrent));
         safeAwait(startHandOffSent);
         ensureGreen(indexName);
-        assertBusy(
-            () -> { assertEquals("Open contexts after shard relocation.", 0, searchService1.getActivePITContexts()); },
-            5,
-            TimeUnit.SECONDS
-        );
+        waitForNoPITContextOnNode(searchNodeCurrent, 5);
 
         // stop the current search node in some cases, this i.e. checks that we close contexts after relocation and don't leak open contexts
         boolean stopFirstNode = randomBoolean();
@@ -591,9 +587,8 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         ensureGreen(indexName);
         logger.info("Search node " + searchNodeCurrent + " excluded.");
 
-        safeSleep(new TimeValue(2, TimeUnit.SECONDS));
         // PIT id should have changed at some point, and all contexts on the source node should be closed
-        assertEquals("Open contexts on node " + searchNodeCurrent + ".", 0L, searchService1.getActivePITContexts());
+        waitForNoPITContextOnNode(searchNodeCurrent, 5);
         assertFalse("pit1 should have changed.", isEquivalentId(pitId1.get(), testDataSetup.pitId1));
 
         thread1Running.set(false);
@@ -1207,7 +1202,7 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
      * transferred during shard relocation. Three rounds of indexing with refresh-only (no flush)
      * accumulate three CCs in a single VBCC. Each PIT is matched to its CC entry when building the handoff.
      */
-    public void testPointInTimeRelocationPitPositionsInBcc() {
+    public void testPointInTimeRelocationPitPositionsInBcc() throws Exception {
         assumeTrue("Requires pit relocation feature flag", PIT_RELOCATION_FEATURE_FLAG.isEnabled());
         // ensure three refresh cycles do not trigger a count-based VBCC upload before relocation
         var nodeSettings = Settings.builder()
@@ -1275,6 +1270,7 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         updateIndexSettings(Settings.builder().put("index.routing.allocation.exclude._name", searchNodeA), indexName);
         ensureGreen(indexName);
         assertThat(internalCluster().nodesInclude(indexName), hasItem(searchNodeB));
+        waitForNoPITContextOnNode(searchNodeA, 5);
 
         AtomicReference<BytesReference> updatedPitFirst = new AtomicReference<>(pitFirst);
         AtomicReference<BytesReference> updatedPitMiddle = new AtomicReference<>(pitMiddle);
@@ -1284,6 +1280,7 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         assertResponse(prepareSearch().setPointInTime(new PointInTimeBuilder(pitFirst)), resp -> {
             // To account for the marker document
             assertHitCount(resp, numDocsFirst + 1);
+            assertFalse("pit should have changed.", isEquivalentId(resp.pointInTimeId(), pitFirst));
             updatedPitFirst.set(resp.pointInTimeId());
         });
         assertResponse(
@@ -1294,6 +1291,7 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         assertResponse(prepareSearch().setPointInTime(new PointInTimeBuilder(pitMiddle)).setSize(1000), resp -> {
             // To account for the marker documents
             assertHitCount(resp, numDocsFirst + numDocsMiddle + 2);
+            assertFalse("pit should have changed.", isEquivalentId(resp.pointInTimeId(), pitMiddle));
             updatedPitMiddle.set(resp.pointInTimeId());
         });
         assertResponse(
@@ -1305,6 +1303,7 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         assertResponse(prepareSearch().setPointInTime(new PointInTimeBuilder(pitLast)).setSize(1000), resp -> {
             // To account for the marker documents
             assertHitCount(resp, numDocsFirst + numDocsMiddle + numDocsLast + 3);
+            assertFalse("pit should have changed.", isEquivalentId(resp.pointInTimeId(), pitLast));
             updatedPitLast.set(resp.pointInTimeId());
         });
         assertResponse(
@@ -1355,15 +1354,7 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         // Relocate all shards from searchNodeA to searchNodeB.
         updateIndexSettings(Settings.builder().put("index.routing.allocation.exclude._name", searchNodeA), indexName);
         ensureGreen(indexName);
-        assertBusy(
-            () -> assertEquals(
-                "Source node should have no active PIT contexts after handoff.",
-                0,
-                internalCluster().getInstance(SearchService.class, searchNodeA).getActivePITContexts()
-            ),
-            5,
-            TimeUnit.SECONDS
-        );
+        waitForNoPITContextOnNode(searchNodeA, 5);
 
         getTelemetryPlugin(searchNodeA).collect();
         getTelemetryPlugin(searchNodeB).collect();
@@ -1402,6 +1393,18 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         );
         assertClosePit(updatedPit1.get(), numberOfShards);
         assertClosePit(updatedPit2.get(), numberOfShards);
+    }
+
+    private void waitForNoPITContextOnNode(String searchNodeName, long maxWaitTimeSeconds) throws Exception {
+        assertBusy(
+            () -> assertEquals(
+                "Node " + searchNodeName + " should have no active PIT contexts.",
+                0,
+                internalCluster().getInstance(SearchService.class, searchNodeName).getActivePITContexts()
+            ),
+            maxWaitTimeSeconds,
+            TimeUnit.SECONDS
+        );
     }
 
     private void assertClosePit(BytesReference pitId, int expectedFreedContexts) {

